### PR TITLE
Logger [4959]

### PR DESCRIPTION
**Pull request by julianbermudez**
This pull request adds a header only logger based on [spdlog](https://github.com/gabime/spdlog).
The logger could be enabled/disabled at compile time using the `PROFILE_LOGGER` CMake option, in case of disabling it has zero footprint on the Agent code.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Agent%20Manual%20Linux/168/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Agent%20Manual%20Windows/171/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Agent%20Manual%20Linux/169/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Agent%20Manual%20Windows/172/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Agent%20Manual%20Linux/177/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Agent%20Manual%20Windows/180/)

File: include/uxr/agent/message/OutputMessage.hpp [hunk 1]
```diff
@@ -12,27 +12,28 @@
...
-#include <uxr/agent/config.hpp>
+
 #include <fastcdr/Cdr.h>
 #include <fastcdr/exceptions/Exception.h>
-#include <iostream>
 
 namespace eprosima {
 namespace uxr {
 
 class OutputMessage
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply include/uxr/agent/message/OutputMessage.hpp]: You are right, I removed the copy and move constructors.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Agent%20Manual%20Linux/190/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Agent%20Manual%20Windows/193/)

**Pull request merged**